ui/appui-react/src/appui-react/widgets/useThisWidget.tsx

```diff
@@ -0,0 +1,102 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import React from "react";
+import { getTabLocation, TabIdContext } from "@itwin/appui-layout-react";
+import { assert } from "@itwin/core-bentley";
+import { useActiveFrontstageDef } from "../frontstage/FrontstageDef";
+import { InternalFrontstageManager } from "../frontstage/InternalFrontstageManager";
+import { UiItemsManager } from "../ui-items-provider/UiItemsManager";
+import type { WidgetState } from "./WidgetState";
+
+/** Interface to be used with the useThisWidget hook. Holds information
+ * about the Widget such as the state and location. Also contains a function to set WidgetState.
+ * @alpha
+ */
+export interface ThisWidget {
```

Comment:
Will we need additional information? Do we plan to add that additional information as optional properties only? Or we could keep this in `alpha` for a while and still do that?

I was thinking about other strategies, i.e. could we consider returning an implicit type from the hook, add additional `@note` that the return type should never be used outside of this hook. But https://www.itwinjs.org/learning/api-support-policies/ doesn't seem to allow such cases.

Reply:
Raphael thought we should start with the included props. I don't think the optional properties will be an issue with this? Since the type is really only used to store information returned from this hook. No information is needed from the user. So if we add additional information the hook should not have a problem returning it, but then the user wouldn't need to use it if they didn't want to. I think we should keep it in alpha for now though

Reply:
Removed the interface and returning object now returns just data. Changed it to the recommendations of this comment in the iTwin team https://teams.microsoft.com/l/message/19:6582ace6c5754414bdab0a95d7420bc6@thread.skype/1703264604580?tenantId=067e9632-ea4c-4ed9-9e6d-e294956e284b&groupId=2fc3ba7e-a05a-4cc0-9411-e9d2d2652ec1&parentMessageId=1703260701203&teamName=iTwin.js&channelName=General&createdTime=1703264604580